ignite/engine/engine.py

```diff
@@ -172,13 +173,15 @@ def _run_once_on_dataset(self):
                 if self.should_terminate:
                     break
 
-            time_taken = time.time() - start_time
-            hours, mins, secs = _to_hours_mins_secs(time_taken)
-            return hours, mins, secs
         except BaseException as e:
             self._logger.error("Current run is terminating due to exception: %s", str(e))
             self._handle_exception(e)
 
+        finally:
+            time_taken = time.time() - start_time
+            hours, mins, secs = _to_hours_mins_secs(time_taken)
+            return hours, mins, secs
```

Comment:
Before this change, when we handled the exception ourselves, `_run_once_on_dataset()` did not return anything, resulting in error, because we tried to unpack `hours, mins, secs = None`. 

Should I refactor this to another PR with a test?

Reply:
@elanmart I think this would be great !